### DISCUSSION TITLE
@import of base, component and utilities not being resolved.

### DISCUSSION THREAD

**Discussion by Caleb-Cohen** [tailwindlabs/tailwindcss#12275] [Help] [answered]
Howdy, 

I'm working on [this branch](https://github.com/Caleb-Cohen/menu/tree/MenuComponent) of an app. It uses Next.JS `12.1.2`, MUI `5.14.3` and Tailwindcss `3.3.3.` I can't seem to find the imports to resolve, as seen below. I've followed [this guide](https://mui.com/material-ui/guides/interoperability/#tailwind-css) on getting Tailwind integrated with MUI.

<img width="1000" alt="Screenshot 2023-10-23 at 11 23 04 PM" src="https://github.com/tailwindlabs/tailwindcss/assets/75268174/b2da49a5-e682-4ec3-b802-568860840a2a">

Here's my `tailwind.config.js`

```TS
/** @type {import('tailwindcss').Config} */
module.exports = {
  content: [
    './pages/**/*.{js,ts,jsx,tsx,mdx}',
    './components/**/*.{js,ts,jsx,tsx,mdx}',
    './app/**/*.{js,ts,jsx,tsx,mdx}',
  ],
  important: '#__next',
  theme: {
    extend: {
      backgroundImage: {
        'gradient-radial': 'radial-gradient(var(--tw-gradient-stops))',
        'gradient-conic':
          'conic-gradient(from 180deg at 50% 50%, var(--tw-gradient-stops))',
      },
    },
  },
  corePlugins: {
    preflight: false,
  },
  plugins: [],
};
```

`globals.css`

```TS
/* globals.css */
@import 'tailwindcss/base';
@import 'tailwindcss/components';
@import 'tailwindcss/utilities';
```

`_app.tsx`

```TS
// These styles apply to every route in the application
import '../styles/globals.css';
import { StyledEngineProvider, ThemeProvider, createTheme } from '@mui/material';
import type { AppProps } from 'next/app';

let theme = createTheme();

function initializeTheme() {
  const rootElement = document.getElementById('__next');

  return createTheme({
    components: {
      MuiPopover: {
        defaultProps: {
          container: rootElement,
        },
      },
      MuiPopper: {
        defaultProps: {
          container: rootElement,
        },
      },
      MuiDialog: {
        defaultProps: {
          container: rootElement,
        },
      },
      MuiModal: {
        defaultProps: {
          container: rootElement,
        },
      },
    },
  });
}

if (typeof document !== 'undefined') {
  theme = initializeTheme();
}

export default function App({ Component, pageProps }: AppProps) {
  return (
    <StyledEngineProvider injectFirst>
      <ThemeProvider theme={theme}>
        <Component {...pageProps} />;
      </ThemeProvider>
    </StyledEngineProvider>
  );
}
```

`postcss.config.js`

```js

module.exports = {
  plugins: {
    tailwindcss: {},
    autoprefixer: {},
  },
};

```

**Comment by wongjn** [comment, marked as answer]
You could consider:
- Using `@tailwind` directives instead of `@import`ing:
  ```diff
  -@import 'tailwindcss/base';
  -@import 'tailwindcss/components';
  -@import 'tailwindcss/utilities';
  +@tailwind base;
  +@tailwind components;
  +@tailwind utilities;
  ```
- Installing [postcss-import](https://github.com/postcss/postcss-import) and including it before Tailwind in `postcss.config.js`, as per [the documentation](https://tailwindcss.com/docs/using-with-preprocessors#build-time-imports):
  ```sh
  npm install postcss-import --save-dev
  ```
  ```diff
   module.exports = {
     plugins: {
  +    'postcss-import': {},
       tailwindcss: {},
       autoprefixer: {},
     }
   }
  ```

**Reply by Caleb-Cohen** [reply]
Hey there! Thanks for the input.

I tried both recommendations together and separately, and it did not resolve the issue.

**Reply by wongjn** [reply]
You may be look at the source map that Chrome shows:

![image](https://github.com/tailwindlabs/tailwindcss/assets/11310624/0507bf5a-7fa9-4022-a17d-e0bd04a3312e)

The *From `localhost/`* in the bottom right of the panel lets you know you are looking at the source file, not the actual file.

Either recommendation above works for me.

**Reply by Caleb-Cohen** [reply]
You are correct that I'm using the sources tab, sorry I had that cropped earlier. 

It's my understanding these imports should resolve into CSS. Is that the case, and if so where is that located within the sources tab?

**Reply by wongjn** [reply]
You should be able to see the "real" CSS delivered to the browser by clicking the underlined text (`localhost/` in the previous screenshot), which indicates a "link" in the devtools. When in development mode, your CSS may be injected inline into the HTML page, not as a file.

**Reply by Caleb-Cohen** [reply]
Taking a look there provides this HTML

```HTML
<!DOCTYPE html>
<html>
    <head>
        <style data-next-hide-fouc="true">
            body {
                display: none
            }
        </style>
        <noscript data-next-hide-fouc="true">
            <style>
                body {
                    display: block
                }
            </style>
        </noscript>
        <meta charSet="utf-8"/>
        <meta name="viewport" content="width=device-width"/>
        <meta name="next-head-count" content="2"/>
        <noscript data-n-css=""></noscript>
        <script defer="" nomodule="" src="/_next/static/chunks/polyfills.js?ts=1698179836829"></script>
        <script src="/_next/static/chunks/webpack.js?ts=1698179836829" defer=""></script>
        <script src="/_next/static/chunks/main.js?ts=1698179836829" defer=""></script>
        <script src="/_next/static/chunks/pages/_app.js?ts=1698179836829" defer=""></script>
        <script src="/_next/static/chunks/pages/index.js?ts=1698179836829" defer=""></script>
        <script src="/_next/static/development/_buildManifest.js?ts=1698179836829" defer=""></script>
        <script src="/_next/static/development/_ssgManifest.js?ts=1698179836829" defer=""></script>
        <script src="/_next/static/development/_middlewareManifest.js?ts=1698179836829" defer=""></script>
        <noscript id="__next_css__DO_NOT_USE__"></noscript>
    </head>
    <body>
        <div id="__next" data-reactroot="">
            <div class="flex flex-col justify-center items-center h-screen w-full border-4 border-indigo-500/100">
                <img src="/images/Takayama_Full_Logo.png" alt="Full Logo"/>
                <span class="mt-4 text-[2vw]">10/21/23</span>
            </div>
            ;
        </div>
        <script src="/_next/static/chunks/react-refresh.js?ts=1698179836829"></script>
        <script id="__NEXT_DATA__" type="application/json">
            {"props":{"pageProps":{}},"page":"/","query":{},"buildId":"development","nextExport":true,"autoExport":true,"isFallback":false,"scriptLoader":[]}
        </script>
    </body>
</html>
```

Even the in-line styles are not working as there's no visible border
<img width="2701" alt="Screenshot 2023-10-24 at 3 54 54 PM" src="https://github.com/tailwindlabs/tailwindcss/assets/75268174/6d23a01f-62c7-4f73-86ca-7a7a89e96583">

Shouldn't this setup look exactly like play.tailwind.css? 

<img width="3155" alt="Screenshot 2023-10-24 at 3 56 33 PM" src="https://github.com/tailwindlabs/tailwindcss/assets/75268174/4931accd-0226-4d70-9bd2-68822121889b">

**Reply by wongjn** [reply]
The border styles *are* being applied:

![image](https://github.com/tailwindlabs/tailwindcss/assets/11310624/564815c4-1025-4f25-b763-73177b58101d)

The border presumably is not showing because you have disabled [preflight](https://tailwindcss.com/docs/preflight).

The border shows in the Tailwind Play because you have **not** disabled preflight there, as evidenced by the snippet of CSS that can be seen in the *Generated CSS* panel:
![image](https://github.com/tailwindlabs/tailwindcss/assets/11310624/b6915edc-cfb3-4efc-8f15-8e1899a6d02a)

And the CSS rules in devtools:

![image](https://github.com/tailwindlabs/tailwindcss/assets/11310624/06c72634-e289-4639-bf06-c389d5c0e027)

that are not present in your project.

**Reply by Caleb-Cohen** [reply]
Is it necessary to use tailwind's preflight to make tailwind work? How can I effectively run both MUI and Tailwind in the same project?

**Reply by wongjn** [reply]
It is not necessary to use tailwind's preflight to make tailwind work. For the border to display, consider applying `border-style: solid` via the `border-solid` class.
